zigpy_xbee/zigbee/application.py

```diff
@@ -186,10 +188,15 @@ async def force_remove(self, dev):
         """Forcibly remove device from NCP."""
         pass
 
-    async def add_endpoint(self, descriptor):
+    async def add_endpoint(self, descriptor: zdo_t.SimpleDescriptor) -> None:
         """Register a new endpoint on the device."""
-        # This is not provided by the XBee API
-        pass
+        self._device.replacement["endpoints"][descriptor.endpoint] = {
```

Comment:
Yes. The idea is that we first fill the endpoint replacement data, which is then utilized here: https://github.com/zigpy/zigpy/blob/cdc2ad2a35c987d34567ceacb42fb14c06ca9c06/zigpy/quirks/__init__.py#L114

Reply:
Well, it is not strictly necessary, the bare minimum to avoid the error is just create an empty endpoint with certain number and just ignore the endpoint descriptor

Reply:
Ah, I see!